src/metric-engine/src/engine/drop.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Drop a metric region
+
+use mito2::engine::MITO_ENGINE_NAME;
+use object_store::util::join_dir;
+use snafu::{OptionExt, ResultExt};
+use store_api::metric_engine_consts::{
+    DATA_REGION_SUBDIR, METADATA_REGION_SUBDIR, PHYSICAL_TABLE_METADATA_KEY,
+};
+use store_api::region_engine::RegionEngine;
+use store_api::region_request::{
+    AffectedRows, RegionDropRequest, RegionOpenRequest, RegionRequest,
+};
+use store_api::storage::RegionId;
+
+use super::MetricEngineInner;
+use crate::error::{
+    CloseMitoRegionSnafu, Error, LogicalRegionNotFoundSnafu, OpenMitoRegionSnafu,
+    PhysicalRegionNotFoundSnafu, Result,
+};
+use crate::metrics::PHYSICAL_REGION_COUNT;
+use crate::{metadata_region, utils};
+
+impl MetricEngineInner {
+    pub async fn drop_region(
+        &self,
+        region_id: RegionId,
+        _req: RegionDropRequest,
+    ) -> Result<AffectedRows> {
+        let data_region_id = utils::to_data_region_id(region_id);
+
+        // enclose the guard in a block to prevent the guard from polluting the async context
+        let (is_physical_region_exist, is_physical_region_busy) = {
+            if let Some(logical_regions) = self
+                .state
+                .read()
+                .unwrap()
+                .physical_regions()
+                .get(&data_region_id)
+            {
+                (true, !logical_regions.is_empty())
+            } else {
+                (false, true)
```

Comment:
(false, false) ?

Reply:
The second is not used. I've add a comment for it